### DISCUSSION TITLE
How to get rid of the square in the bottom right corner of a treeview?

### DISCUSSION THREAD

**Discussion by YoriMirus** [AvaloniaUI/Avalonia#7802] [Q&A] [answered]
Hello. I'm making a treeview with a custom scrollbar for my project and noticed when both of the scrollbars are visible, there is this small white square in the bottom right of the treeview.

I can't figure out how to remove it, changing background to transparent doesn't do anything with it. I looked around in the scrollbar style but I can't see anything that could cause this.

Here is the template I'm using, I only changed a few corner radiuses and changed scrollbar's TrackRect's Fill to Transparent, however even the default style has that square.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ScrollBar.xaml

Anyone knows what I'm supposed to change to get rid of it?
Here is an image showing the square:

![image](https://user-images.githubusercontent.com/66534752/158181067-32bace52-60a8-491f-8b1f-1633e24eed22.png)

(this is inside a test solution I have made just to test the custom scrollbar)

**Comment by pr8x** [comment, marked as answer]
You can easily find out which part of the control it is by using the DevTools. Just open them (with F12) and click Ctrl+Shift over the area you want to inspect. It should then highlight the item. You can then style its background.

**Reply by YoriMirus** [reply]
Thank you for the response. It seems like the culprit is some kind of panel inside the scrollviewer, which is inside the treeview. Unfortunately I don't know much about styling, so I'm not sure how to fix this. Is there a way to change the panel's background without having to copy the whole 160 line ScrollViewer.xaml just to replace one property?

Here is the panel:

![image](https://user-images.githubusercontent.com/66534752/158232771-187d9d45-d169-445e-88c0-2e8a739a9999.png)

**Reply by jp2masa** [reply]
You should be able to override these styles:

https://github.com/AvaloniaUI/Avalonia/blob/825a0247110913f8570dce32b7854f1069d55f38/src/Avalonia.Themes.Fluent/Controls/ScrollViewer.xaml#L73-L84

**Reply by YoriMirus** [reply]
Thank you very much! That indeed worked. In case anyone has the same problem as me here is my solution:

![image](https://user-images.githubusercontent.com/66534752/158253809-ad02692f-bee2-40e6-b45e-45faabc47fdf.png)


If you change the opacity value to 0, then there will be a short blinking animation playing when the scrollviewer expands, overriding that transition and making its duration 0ms long will make that animation invisible, thus no white square remains.

Just put this in a style folder and include it in the treeview.
